src/xtopcom/xcommon/xtop_event.h

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2017-2021 Telos Foundation & contributors
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#pragma once
+
+#include "xcommon/xtop_log.h"
+
+#include <string>
+#include <vector>
+
+NS_BEG2(top, common)
+
+enum enum_event_data_type {
+    enum_event_string = 1,
+    enum_event_uint64 = 2,
+    enum_event_address = 3,
+};
+
+class argument_t {
+public:
+    argument_t() = default;
+    argument_t(argument_t const &) = default;
+    argument_t & operator=(argument_t const &) = default;
+    argument_t(argument_t &&) = default;
+    argument_t & operator=(argument_t &&) = default;
+    ~argument_t() = default;
+
+    argument_t(std::string const & str_data, enum_event_data_type const & type, bool const & indexed) : str_data(str_data), type(type), indexed(indexed){};
```

Comment:
only std::string (or user defined struct or class) is recommended to add `const &`.  enum, boolean and other fundamental types don't need to add `const &`.